### PR TITLE
Update changelog for 1.2.3 release

### PR DESCRIPTION
**Pull request by meltsufin**

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -5,6 +5,56 @@ https://spring.io/projects/spring-cloud-gcp[Spring Cloud GCP] is a set of integr
...
 
+== 1.2.3.RELEASE (2020-05-29)
+
+=== General
+
+* Upgrade to latest libraries bom and sql socket factory (https://github.com/spring-cloud/spring-cloud-gcp/issues/2373[#2373])
+* Make transaction managers conditional on enabled flag for Spanner and Datastore (https://github.com/spring-cloud/spring-cloud-gcp/issues/2376[#2376])
+
+=== Secret Manager 
+
```
Comment (dzou) [inline CHANGELOG.adoc:16]: I would add a more explicit statement that there are breaking changes between 1.2.2 and 1.2.3 for secret manager.

Maybe somehting like:

"Version 1.2.3.RELEASE introduces breaking changes to using Secret Manager as a Spring Boot property source. Please see the [reference documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-gcp/1.2.3.RELEASE/reference/html/#secret-manager-property-source) for the updated syntax for how to specify secrets in your application's `.properties` files."

Comment (meltsufin) [reply CHANGELOG.adoc:16]: Done.

**Comment by meltsufin** [comment]
> We have to describe the breaking change in logback logging format -- from plaintext to JSON.

Good catch! Added a note.

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -5,6 +5,64 @@ https://spring.io/projects/spring-cloud-gcp[Spring Cloud GCP] is a set of integr
...
+* Remove the version property in secret manager (https://github.com/spring-cloud/spring-cloud-gcp/issues/2270[#2270])
+* Secret manager template with project (https://github.com/spring-cloud/spring-cloud-gcp/issues/2283[#2283]) (https://github.com/spring-cloud/spring-cloud-gcp/issues/2284[#2284])
+* Create protocol for specifying secrets' project and versions (https://github.com/spring-cloud/spring-cloud-gcp/issues/2302[#2302])
+* Add secret manager autoconfigure property (https://github.com/spring-cloud/spring-cloud-gcp/issues/2363[#2363])
+
+=== Pub/Sub 
+
+* Suppress exception in Pub/Sub adapter in AUTO_ACK and MANUAL modes (https://github.com/spring-cloud/spring-cloud-gcp/issues/2319[#2319])
+* Detailed docs on acking/nacking in Spring Integration (https://github.com/spring-cloud/spring-cloud-gcp/issues/2078[#2078])
```
Comment (elefeint) [inline CHANGELOG.adoc:33]: I don't think we need to add documentation changes to changelog.

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -5,6 +5,64 @@ https://spring.io/projects/spring-cloud-gcp[Spring Cloud GCP] is a set of integr
...
+* Add child collection to the entity class in Firestore sample (https://github.com/spring-cloud/spring-cloud-gcp/issues/2388[#2388])
+
+=== Vision
+
+* Allow users to provide the ImageContext in CloudVisionTemplate (https://github.com/spring-cloud/spring-cloud-gcp/issues/2286[#2286])
+
+=== Firebase Security
+
+* Make Firebase Security Autoconfiguration conditional (https://github.com/spring-cloud/spring-cloud-gcp/issues/2258[#2258])
+
```
Comment (elefeint) [inline CHANGELOG.adoc:65]:
```suggestion
Thank you to the contributors from our user community: @eddumelendez, @mzeijen,@s13o, @acet, @guillaumeblaquiere
```

File: CHANGELOG.adoc [hunk 4]
```diff
@@ -5,6 +5,64 @@ https://spring.io/projects/spring-cloud-gcp[Spring Cloud GCP] is a set of integr
...
+NOTE: This version introduced several breaking changes to Secret Manager property source.
+Please see the https://cloud.spring.io/spring-cloud-static/spring-cloud-gcp/1.2.3.RELEASE/reference/html/#secret-manager-property-source)[reference documentation] for the new way for accessing secrets as properties.
+
+* Remove the version property in secret manager (https://github.com/spring-cloud/spring-cloud-gcp/issues/2270[#2270])
+* Secret manager template with project (https://github.com/spring-cloud/spring-cloud-gcp/issues/2283[#2283]) (https://github.com/spring-cloud/spring-cloud-gcp/issues/2284[#2284])
+* Create protocol for specifying secrets' project and versions (https://github.com/spring-cloud/spring-cloud-gcp/issues/2302[#2302])
+* Add secret manager autoconfigure property (https://github.com/spring-cloud/spring-cloud-gcp/issues/2363[#2363])
+
+=== Pub/Sub 
+
```
Comment (elefeint) [inline CHANGELOG.adoc:31]:
```suggestion
* New async pull methods in `Pub/Sub Template`, as well as fully asynchronous `PubSubReactiveFactory.poll()` (https://github.com/spring-cloud/spring-cloud-gcp/pull/2227[#2227])
```

File: CHANGELOG.adoc (hunk 2 above)
Comment (meltsufin) [reply CHANGELOG.adoc:33]: I thought these were major enough to give them a shoutout, but OK. :)

File: CHANGELOG.adoc (hunk 4 above)
Comment (meltsufin) [reply CHANGELOG.adoc:31]: I don't know how I missed this one. Good catch!

File: CHANGELOG.adoc [hunk 5]
```diff
@@ -62,7 +61,7 @@ Please see the https://cloud.spring.io/spring-cloud-static/spring-cloud-gcp/1.2.
 === Firebase Security
 
 * Make Firebase Security Autoconfiguration conditional (https://github.com/spring-cloud/spring-cloud-gcp/issues/2258[#2258])
-
+Thank you to the contributors from our user community: @eddumelendez, @mzeijen,@s13o, @acet, @guillaumeblaquiere
```
Comment (elefeint) [inline CHANGELOG.adoc:64]:
```suggestion
Thank you to the contributors from our user community: @eddumelendez, @mzeijen, @s13o, @acet, @guillaumeblaquiere
```

**Pull request merged**